api/src/main/java/com/datatorrent/api/StorageAgent.java

```diff
@@ -94,4 +94,24 @@
     public void setApplicationAttributes(AttributeMap map);
   }
 
+  interface AsyncStorageAgent
+  {
+    /**
+     * Make checkpoint for given windowID final. i.e after this method returns,
+     * the checkpoint is accessible for recovery.
+     *
+     * @param operatorId
+     * @param windowId
+     * @throws IOException
+     */
+    void finalize(int operatorId, long windowId) throws IOException;
+
+    /**
+     * Check if StorageAgent is configured to take synchronous checkpoints.
+     *
+     * @return true if StorageAgent is configured to take synchronous checkpoints.
+     * @return false otherwise.
+     */
+    boolean isSyncCheckpoint();
```

Comment:
On second thoughts.. it looks like the sync checkpoint flag is to force a sync behavior on the async agent and is being used by tests. I don't think this should be part of the interface. Second, I am not sure if this functionality is needed as the tests seem to be creating the AsyncFSStorageAgent and setting this flag immediately, in that case why not just create a FSStorageAgent instead?  If this functionality is needed for testing, it can be an internal artifice. One way to do this, for example, is to define a new internal interface, not in api, and add this method there. You could call this interface SyncCompatibleAgent from example.